### ISSUE TITLE
Integrate YouTube Music Results

### ISSUE THREAD

**Issue by alaskanpuffin** [nukeop/nuclear#922]
I'm not sure how feasible this would be, but I feel like integration with YouTube Music would make nuclear significantly better. YouTube Music (music.youtube.com) offers higher quality versions of most songs, such as no intro/outro sections and in my opinion better overall audio quality.

I really like the idea of Nuclear, but I always end up going back to YouTube Music as Nuclear just doesn't have the same audio quality, which is likely caused by the lower quality YouTube video versions.

**Comment by HaiDang666** [comment]
nice suggestion

**Comment by alaskanpuffin** [comment]
Wow, that might be one of the fastest responses I have seen. Thanks for looking into it!

**Comment by HaiDang666** [comment]
Just quick research on npm, there are few packages for music.youtube. I think we can use them

**Comment by nukeop** [comment]
I don't have a google account, so I can't work on it. Not sure if it can be used without registering Nuclear to get an API key. Maybe we can require users to provide their own api keys to avoid using a single key for everyone.

**Comment by OhShitNation** [comment]
This would if we could download the playlists from youtube too or my last.fm playlist

**Comment by nukeop** [comment]
Does last.fm let you create playlists now? We could add a way to import and sync them from there.

**Comment by OhShitNation** [comment]
Yes; it would be nice to be able to download the Playlist as a whole but I still would like to be able to import & download my playlists from Youtube as most of the songs I have can't be played or found I'm looking into youtube-dl and making it a plugin but frankly I can bearly operate a command prompt so thats going to be interesting. Lol

**Comment by HaiDang666** [comment]
@OhShitNation currently the youtube playlist function is broken, you can use this lib (https://github.com/HaiDang666/extract-sptf-playlist) to extract your playlists manually into file then add them back to nuclear

**Comment by HaiDang666** [comment]
note: some countries is not available with youtube music

**Comment by nukeop** [comment]
We need to find out if using yt music in Nuclear would not violate their policies, both for us (risk of legal action) and for the users (risk of getting banned). Sometimes alternative clients are forbidden, or clients are restricted in what they can do. We also need to figure out if we will need an API key.

**Comment by 07416** [comment]
> I'm not sure how feasible this would be, but I feel like integration with YouTube Music would make nuclear significantly better. YouTube Music (music.youtube.com) offers higher quality versions of most songs, such as no intro/outro sections and in my opinion better overall audio quality.

The main point being that labels and artists don't generally upload full albums to Youtube, especially as playlists (some publish the full album as a single "video") — it's all user rips.

**Comment by 07416** [comment]
> We need to find out if using yt music in Nuclear would not violate their policies, both for us (risk of legal action) and for the users (risk of getting banned). Sometimes alternative clients are forbidden, or clients are restricted in what they can do. We also need to figure out if we will need an API key.

[Newpipe](https://newpipe.net/) has supported Youtube Music for years, including download support.

**Comment by nukeop** [comment]
> > I'm not sure how feasible this would be, but I feel like integration with YouTube Music would make nuclear significantly better. YouTube Music (music.youtube.com) offers higher quality versions of most songs, such as no intro/outro sections and in my opinion better overall audio quality.
> 
> The main point being that labels and artists don't generally upload full albums to Youtube, especially as playlists (some publish the full album as a single "video") — it's all user rips.

Not true, there's a ton of albums and songs uploaded by artists and labels.
Example: https://www.youtube.com/user/zappa/playlists?view=71&sort=dd&shelf_id=0

**Comment by N1N74** [comment]
> > > I'm not sure how feasible this would be, but I feel like integration with YouTube Music would make nuclear significantly better. YouTube Music (music.youtube.com) offers higher quality versions of most songs, such as no intro/outro sections and in my opinion better overall audio quality.
> > 
> > 
> > The main point being that labels and artists don't generally upload full albums to Youtube, especially as playlists (some publish the full album as a single "video") — it's all user rips.
> 
> Not true, there's a ton of albums and songs uploaded by artists and labels.
> Example: https://www.youtube.com/user/zappa/playlists?view=71&sort=dd&shelf_id=0

+1, ytm is intended as music streaming platform as much as gpm was before it was discontinued in favor of ytm. I've been using ytm for I think a year now, and never had any issue with missing music especially considering I managed to import over 3k songs from my library. Even obscure indie artists and soundtracks are listed. YTM search plugin would be perfect considering ytm is able to differentiate music from music video. (there are some quirks like music listed in album replaced by MV but the song themselves still individually exist on ytm on search, ironically linking to same the album)

**Comment by BobIsMyManager** [comment]
> We need to find out if using yt music in Nuclear would not violate their policies, both for us (risk of legal action) and for the users (risk of getting banned). Sometimes alternative clients are forbidden, or clients are restricted in what they can do. We also need to figure out if we will need an API key.

By the way, there is already a youtube music frontend called [Beatbump](https://github.com/snuffyDev/Beatbump) which works really well, so it is clearly possible to do without user issues and (probably) legal issues.

**Comment by 07416** [comment]
> By the way, there is already a youtube music frontend called [Beatbump](https://github.com/snuffyDev/Beatbump) which works really well, so it is clearly possible to do without user issues and (probably) legal issues.

@BobIsMyManager Is this a web wrapper?

**Comment by BWPanda** [comment]
If it helps, [RiMusic](https://github.com/fast4x/RiMusic) for Android streams from YouTube Music without needing an account/API key from the user.

**Comment by nukeop** [comment]
We can do that too, with the quality offered to anonymous users. But we can't offer logging in yet.

**Comment by nukeop** [comment]
This issue is from an older version of Nuclear that has been replaced with a complete rewrite. None of the old issues apply to the new codebase.

The old code is preserved on the [legacy/electron](https://github.com/nukeop/nuclear/tree/legacy/electron) branch. If you have feedback about the new version, head to [Discussions](https://github.com/nukeop/nuclear/discussions).
